### ISSUE TITLE
True TV colours - request/suggestion

### ISSUE THREAD

**Issue by andrew-davie** [stella-emu/stella#538]
I was rather surprised to see a video of Sokoboo running on an NTSC television where the colours on the TV were *significantly* different than on Stella. Yes I know NTSC is "never the same colour" and that there is a "hue" knob of some sort. But it wasn't just the colour, it was the intensity too.

I'm wondering if it might be possible to calculate the ACTUAL colours (or signal, more to the point) sent in a NTSC atari to the TV?  That is, put a scope on the signal, perhaps, and interpret the colour-burst?  Or even dig up the hardware specs of the chip doing the colour burst data?  And then, offering a "hue" control in Stella?

Basically I'm having a real hard time figuring out what should be the colours/intensities to use when a) every emulator is basically different, and b) Stella is different to the TV pictures I saw and c) different TV pictures also look different.

**Comment by thrust26** [comment]
Stella already offers user defined palettes. So you can define a number of palettes and then find the best compromise for your colors chosen. Maybe you find someone on AtariAge who volunteers to provide the necessary values.

Simulating the various combined settings and inaccuracies of the analog hardware in Stella seems like overkill to me.

**Comment by f21red** [comment]
I guess an EE person at AtariAge would have to measure the raw color signals or voltages from the console. Then convert to RGB.

I know some NES emulators support this feature.
https://github.com/SourMesen/Mesen/blob/master/Core/RawVideoFilter.cpp

And I think TmEE of nesdev logged console voltages for Sega consoles like SMS and Megadrive.

But no one I've heard has yet done it for Atari or NEC consoles.

**Comment by sa666666** [comment]
The latest version of z26 (4.07 I believe) is reported to have something like this (I haven't tried it out).  They say it comes from the latest MAME code.  So perhaps we can look there for ideas.  All these projects are now under GPLv2, so we're free to re-use code if we like.

**Comment by thrust26** [comment]
Looking into this now...

**Comment by thrust26** [comment]
I looked into z26 and MAME code. Overall they are doing the same. While MAME strictly follows a function, z26 does a pretty big, manual correction of base color values (see line 33 in attached file).

The comments says:
> // phase corrections since the 2600 doesn't generate perfect sine waves
// Set corrections for chroma1 and chromaF to be the same to get
// phase shift numbers to match mame numbers.

Does anyone have any more information about these imperfect sine waves? 

[palette.zip](https://github.com/stella-emu/stella/files/4579976/palette.zip)

**Comment by thrust26** [comment]
Implemented with dd7dbae
